### PR TITLE
Make the 'register_runner' label shorter and more idempotent

### PR DESCRIPTION
**Pull request by enescakir**
The 'register_runner' label is one of the most time-consuming labels. It has low idempotency; it may fail or the respirate may exit at an undesired line. We have encountered several incidents because of this.

To avoid storing the 'encoded_jit_config', we initiate an API call and an SSH connection under the same label. However, we can separate the part where we check the status of the runner-script. When the label has a high line count, it becomes more difficult to debug because it's challenging to locate the corresponding line. Additionally, we have a status check at the 'wait' label, which I have now combined with the other check.

### CODE REVIEW

**Review by fdr — APPROVED** [review]
I like how the code got shorter.

File: prog/vm/github_runner.rb [hunk 1]
```diff
@@ -155,50 +155,50 @@ def before_run
...
+    github_client.delete("/repos/#{github_runner.repository_name}/actions/runners/#{runner[:id]}")
+    nap 5
+  end
 
+  label def wait
     case vm.sshable.cmd("systemctl show -p SubState --value #{SERVICE_NAME}").chomp
-    when "exited", "running"
-      hop_wait
+    when "exited"
+      github_runner.incr_destroy
```
Comment (furkansahin) [inline prog/vm/github_runner.rb:194]: why don't you `hop_destroy`?

Comment (enescakir) [reply prog/vm/github_runner.rb:194]: "before_run" might have a logic. For example it finalizes the billing records at vm nexus.

Comment (furkansahin) [reply prog/vm/github_runner.rb:194]: This is getting out of hand, we should keep that logic consistent. For now, you don't have anything in `before_run`, it just allows us to destroy from any state. So, maybe, we should not allow `hop_destroy` ever.

I'm good with your change here.

**Review by furkansahin — APPROVED** [review]

File: prog/vm/github_runner.rb (hunk 1 above)
Comment (enescakir) [reply prog/vm/github_runner.rb:194]: I've reviewed the codebase, and all calls to `hop_destroy` are located within the `before_run` function, with the exception of `Prog::Vnet::SubnetNexus`. I need to further consider how we can restrict the use of `hop_destroy` to only the `before_run` function.

File: prog/vm/github_runner.rb [hunk 2]
```diff
@@ -155,50 +155,50 @@ def before_run
...
+    # having to store the encoded_jit_config.
+    command = "./actions-runner/run.sh --jitconfig #{response[:encoded_jit_config].shellescape}"
+    vm.sshable.cmd("sudo systemd-run --uid runner --gid runner --working-directory '/home/runner' --unit #{SERVICE_NAME} --remain-after-exit -- #{command}")
+
+    hop_wait
+  rescue Octokit::Conflict => e
+    unless e.message.include?("Already exists")
+      raise e
+    end
+    # If the runner already exists and the model lacks a 'runner_id'
```
Comment (byucesoy) [inline prog/vm/github_runner.rb:175]: I think this comment is not entirely correct anymore as we don't check for runner_id above as before.

**Review by byucesoy — APPROVED** [review]

File: prog/vm/github_runner.rb (hunk 2 above)
Comment (enescakir) [reply prog/vm/github_runner.rb]: Yeah. Good catch. I rewrote it.

**Pull request merged**